refactor: Move attribute and routing-forms code from `app-store`/`lib` to `features`

hbjORbj opened the pull request:
## What does this PR do?

This PR reorganizes code to establish proper module boundaries and prevent circular dependencies in the Cal.com codebase.

### Why This Change Is Needed

The core problem is **circular dependency prevention**: The `app-store` package should not import from `features`, but `features` can import from `app-store`. Previously, routing-forms logic in `app-store` was importing from `features`, creating potential circular dependencies that could cause build issues and make the codebase harder to maintain.

Additionally, **feature-specific code was scattered across generic directories**. Attribute repositories and services lived in `packages/lib/` rather than alongside the feature code they support. This made it harder to understand the codebase and maintain clear ownership boundaries.

### Impact

- **Eliminates circular dependency risk** between `app-store` and `features` packages
- **Improves code discoverability** by co-locating feature code (attributes now live in `features/attributes/`, routing-forms logic in `features/routing-forms/`)
- **Sets architectural precedent** for keeping feature code self-contained within `features/` directories
- **No functional changes** - this is purely a refactoring of file locations and import paths

### What Was Moved

1. **Attribute repositories and services**: `PrismaAttributeRepository`, `PrismaAttributeToUserRepository`, `PrismaAttributeOptionRepository`, and related services moved from `packages/lib/server/repository/` and `packages/lib/service/attribute/server/` to `packages/features/attributes/`

2. **Routing forms logic**: `findTeamMembersMatchingAttributeLogic` and `handleResponse` moved from `packages/app-store/` to `packages/features/routing-forms/lib/`

3. **Query resolution utilities**: `resolveQueryValue.ts` moved from `packages/lib/raqb/` to `packages/app-store/routing-forms/lib/`

4. **Type consolidation**: Types moved to `packages/app-store/routing-forms/types/types.d.ts` and `packages/lib/service/attribute/types.d.ts` was deleted

## Mandatory Tasks (DO NOT REMOVE)

- [x] I have self-reviewed the code (A decent size PR without self-review might be rejected).
- [x] N/A - I have updated the developer docs in /docs if this PR makes changes that would require a [documentation change](https://cal.com/docs). If N/A, write N/A here and check the checkbox.
- [x] N/A - I confirm automated tests are in place that prove my fix is effective or that my feature works.

hbjORbj committed:
mv

hbjORbj committed:
wip

hbjORbj committed:
wip

hbjORbj committed:
wip

hbjORbj committed:
Merge branch 'main' into refactor/attribute-repos

pedroccastro commented on packages/lib/service/attribute/types.d.ts:
```diff
@@ -16,3 +17,30 @@ export type AttributeOptionAssignment = AttributeToUser & {
   };
 };
+
+export type Attribute = {
```
Hey! Quick note: Attribute is both imported from Prisma (line 1) and defined locally (line 22). If both are needed, it might be better to rename one

hbjORbj committed:
Merge branch 'main' into refactor/attribute-repos

hbjORbj committed:
Merge remote-tracking branch 'origin/main' into refactor/attribute-repos

hbjORbj committed:
wip

hbjORbj committed:
Merge remote-tracking branch 'origin/main' into refactor/attribute-repos

hbjORbj committed:
plural

devin-ai-integration[bot] committed:
fix: resolve TypeScript type errors for attribute refactoring

- Add TransformedAttributeOption type to handle transformed contains field
- Update imports to use routing-forms Attribute type where needed
- Fix getValueOfAttributeOption to use TransformedAttributeOption type
- Update AttributeOptionValueWithType to use TransformedAttributeOption
- Fix pre-existing lint warnings (any -> unknown, proper type casting)

Co-Authored-By: benny@cal.com <sldisek783@gmail.com>

hbjORbj committed:
fix

hbjORbj committed:
cleanup

hbjORbj committed:
cleanup

hbjORbj committed:
fix

hbjORbj committed:
fix

hbjORbj committed:
fix

hbjORbj committed:
fix

hbjORbj committed:
Merge branch 'main' into refactor/attribute-repos

hbjORbj committed:
fix

hbjORbj committed:
Merge remote-tracking branch 'origin/refactor/attribute-repos' into refactor/attribute-repos

devin-ai-integration[bot] committed:
fix: add explicit type annotation to reduce callback in getAttributes.ts

Co-Authored-By: benny@cal.com <sldisek783@gmail.com>

devin-ai-integration[bot] committed:
Merge branch 'main' into refactor/attribute-repos

hbjORbj committed:
Merge remote-tracking branch origin/main into refactor/attribute-repos

devin-ai-integration[bot] committed:
fix: use RouterOutputs from @calcom/trpc/react for consistent type inference

- Update AppPage.tsx to use RouterOutputs from @calcom/trpc/react instead of inferRouterOutputs<AppRouter>
- Update MultiDisconnectIntegration.tsx to use the same RouterOutputs type source
- Add eslint-disable comments for pre-existing warnings (useEffect deps, img elements)

Co-Authored-By: benny@cal.com <sldisek783@gmail.com>

devin-ai-integration[bot] committed:
fix: explicitly type attribute property in AttributeOptionAssignment

The attribute property from Pick<AttributeOption, 'attribute'> was typed as
'unknown' because Prisma relations don't have explicit types when picked.
This explicitly defines the attribute shape with id, type, and isLocked
properties that are used in assignValueToUser.ts.

Co-Authored-By: benny@cal.com <sldisek783@gmail.com>

devin-ai-integration[bot] committed:
fix: remove unused assignedUsers from AttributeOptionAssignment Pick

The assignedUsers property was not being used in the code but was required
by the Pick type, causing a type mismatch when the actual data from the
database query didn't include it.

Co-Authored-By: benny@cal.com <sldisek783@gmail.com>

devin-ai-integration[bot] committed:
fix: update import path for AttributeOptionAssignment and BulkAttributeAssigner types

The types.d.ts file was moved from packages/lib/service/attribute/ to
packages/app-store/routing-forms/types/. Updated the import path in utils.ts
to point to the new location.

Co-Authored-By: benny@cal.com <sldisek783@gmail.com>

hbjORbj committed:
fix

hbjORbj committed:
Merge branch 'main' into refactor/attribute-repos
